cmd/root_test.go

```diff
@@ -369,3 +370,92 @@ func TestArgParsing_ExperimentalMetadataPrefetchFlag_Failed(t *testing.T) {
 		})
 	}
 }
+
+func TestArgsParsing_GCSAuthFlags(t *testing.T) {
+	wd, err := os.Getwd()
+	require.Nil(t, err)
+	tests := []struct {
+		name           string
+		args           []string
+		expectedConfig *cfg.Config
+	}{
+		{
+			name: "Test gcs auth flags.",
+			args: []string{"gcsfuse", "--anonymous-access", "--key-file=key.file", "--reuse-token-from-url", "--token-url=www.abc.com", "abc", "pqr"},
+			expectedConfig: &cfg.Config{
+				GcsAuth: cfg.GcsAuthConfig{
+					AnonymousAccess:   true,
+					KeyFile:           cfg.ResolvedPath(path.Join(wd, "key.file")),
+					ReuseTokenFromUrl: true,
+					TokenUrl:          "www.abc.com",
+				},
+			},
+		},
+		{
+			name: "Test default gcs auth flags.",
+			args: []string{"gcsfuse", "abc", "pqr"},
+			expectedConfig: &cfg.Config{
+				GcsAuth: cfg.GcsAuthConfig{
+					AnonymousAccess:   false,
+					KeyFile:           "",
+					ReuseTokenFromUrl: true,
+					TokenUrl:          "",
+				},
+			},
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			var gotConfig *cfg.Config
+			cmd, err := NewRootCmd(func(cfg *cfg.Config, _ string, _ string) error {
+				gotConfig = cfg
+				return nil
+			})
+			require.Nil(t, err)
+			cmd.SetArgs(tc.args)
+
+			err = cmd.Execute()
+
+			if assert.NoError(t, err) {
+				assert.Equal(t, tc.expectedConfig.GcsAuth, gotConfig.GcsAuth)
+			}
+		})
+	}
+}
+
+func TestArgsParsing_GCSAuthFlagsThrowsError(t *testing.T) {
+	tests := []struct {
+		name           string
+		args           []string
+		expectedConfig *cfg.Config
+	}{
+		{
+			name: "Invalid value for anonymous-access flag",
+			args: []string{"gcsfuse", "--anonymous-access=a", "abc", "pqr"},
+		},
+		{
+			name: "Invalid value for reuse-token-from-url flag",
+			args: []string{"gcsfuse", "--reuse-token-from-url", "b", "abc", "pqr"},
+		},
+		{
+			name: "Invalid value for token-url flag",
+			args: []string{"gcsfuse", "--token-url=a_b://abc", "abc", "pqr"},
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			cmd, err := NewRootCmd(func(cfg *cfg.Config, _ string, _ string) error {
```

Comment:
Nit: type could be deleted more so because we don't care about the parameter anyway.
```suggestion
			cmd, err := NewRootCmd(func(cfg *cfg.Config, _, _ string) error {
```